src/Build.UnitTests/Parser_Tests.cs

```diff
@@ -527,5 +527,73 @@ public void VerifyNoWarningForOrder()
             // Make sure the log contains the correct strings.
             Assert.DoesNotContain("MSB4130:", ml.FullLog); // "No need to warn for this expression - ($(a) == 1 or $(b) == 2) and $(c) == 3."
         }
+
+        // see https://github.com/dotnet/msbuild/issues/5436
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void SupportItemDefinationGroupInWhenOtherwise(bool context)
+        {
+            var projectContent = $@"
+                <Project ToolsVersion= `msbuilddefaulttoolsversion` xmlns= `msbuildnamespace`>
+                    <Choose>
```

Comment:
This might be out of topic but the only child element that doesn't support `Condition` attribute is `Choose`. Is it on purpose? I find myself, on some occasions, **needing** to add a conditional attribute to the `Choose` element itself.

If I don't have that option, the code kinda becomes error prone and cumbersome.

For Example:

```xml
<Choose Condition="'$(RootCheck)' == 'true'">
  <When Condition="'$(ActualCheck)' == 'value1'">
  </When>
  <When Condition="'$(ActualCheck)' == 'value2'">
  </When>
  <Otherwise>
  </Otherwise>
</Choose>
```

is better, less repetitive and less error prone than the following...

```xml
<Choose>
  <When Condition="'$(RootCheck)' == 'true' AND '$(ActualCheck)' == 'value1'">
  </When>
  <When Condition="'$(RootCheck)' == 'true' AND '$(ActualCheck)' == 'value2'">
  </When>
  <Otherwise Condition="'$(RootCheck)' == 'true'">
  </Otherwise>
</Choose>
```

Reply:
This test is for the ItemDefinitionGroup supported under choose/when and choose/otherwise, and not for the choose when or otherwise condition expression testing. So, didn't add condition expression for the Choose.

Reply:
>  only child element that doesn't support Condition attribute is Choose. Is it on purpose?

Reaching back over 15 years .. no particular reason, it just didn't occur to anyone. I don't see why it couldn't be added if they want it.

Reply:
> only child element that doesn't support Condition attribute is Choose. Is it on purpose?

This should be on purpose from this test case    https://github.com/dotnet/msbuild/blob/8872ed650643d12340fb22ece223c1aacf5a30a0/src/Build.OM.UnitTests/Construction/ProjectChooseElement_Tests.cs#L43-L61

Reply:
@JaynieBai Nice Find... Thanks for looking it up.

Seems on purpose but WHY? Instead of supporting it, they made sure it didn't. Still, WHY? I'm puzzled.

I need this badly. Any chance that this can be allowed?

Reply:
I'm not sure either.  @Forgind @rainersigwald Do you know anything about this?

Reply:
I said why 🙂 I implemented it. It just didn't occur to me at the time that it would be useful. Today I see no reason why it can't be added.

Reply:
> It just didn't occur to me at the time that it would be useful.

So, were you one of devs who worked on MSBuild? 😲 I thought you were just speaking for the team. Sorry If I misunderstood you. 😥

Reply:
Ha.. no problem. Yes, but I don't speak for the current team. I work on ASPNET now. My knowledge is historical.

Reply:
@Nirmal4G, as far as I'm concerned, you're free to file a bug to add conditions as an option. My first impulse is that we'd treat it the same as ItemDefinitionGroups: if you want to implement it, and you do it properly, we can take it, but we aren't likely to prioritize it. But file the bug first so we can have an official team opinion.